### ISSUE TITLE
About Blanced_L1_Loss

### ISSUE THREAD

**Issue by Taylor-X76** [open-mmlab/mmdetection#2483]
In mmdetection-master\mmdet\models\losses\balanced_l1_loss.py 

`@weighted_loss
def balanced_l1_loss(pred,
                     target,
                     beta=1.0,
                     alpha=0.5,
                     gamma=1.5,
                     reduction='mean'):
    assert beta > 0
    assert pred.size() == target.size() and target.numel() > 0

    diff = torch.abs(pred - target)
    b = np.e**(gamma / alpha) - 1
    loss = torch.where(
        diff < beta, alpha / b *
        (b * diff + 1) * torch.log(b * diff / beta + 1) - alpha * diff,
        gamma * diff + gamma / b - alpha * beta)

    return loss
`
The loss value should not be divided by beta（torch.log(b * diff / beta + 1)）
I've corrected it to this.

`def balanced_l1_loss(pred,
                     target,
                     beta=1.0,
                     alpha=0.5,
                     gamma=1.5,
                     reduction='mean'):
    assert beta > 0
    assert pred.size() == target.size() and target.numel() > 0

    diff = torch.abs(pred - target)
    b = (np.e**(gamma / alpha) - 1) / beta
    loss = torch.where(
        diff < beta, alpha / b *
        (b * diff + 1) * torch.log(b * diff + 1) - alpha * diff,
        gamma * diff + gamma / b - alpha * beta)

    return loss`

**Comment by Taylor-X76** [comment]
If the wrong loss function is used for training, then in this case（'beta=1.0/9.0,alpha=0.5,gamma=1.0;'）, the network gradient will increase as the regression error decreases.

**Comment by Taylor-X76** [comment]
https://github.com/open-mmlab/mmdetection/blob/a1c3aa4908a6cd32fcf46691fc20ff855334dae2/mmdet/models/losses/balanced_l1_loss.py#L23

**Comment by OceanPang** [comment]
@Taylor-X76 Hi, the term `beta` here is used to adjust the turning point which is the same as in `smooth l1 loss`. 
We cannot remove the beta here directly. If you remove it, the loss function and gradient function are not continuous anymore. Indeed as you said, the gradient curve here will change with the different beta, thus you need to choose another alpha and gamma to make it smooth or even sharper.

**Comment by Taylor-X76** [comment]
Thanks for your response.
